### PR TITLE
Fix vector store logic and refactor audience parameter

### PR DESCRIPTION
**Pull request by KennyZhang1**
## Description

This PR makes several improvements to the vector_store settings for LanceDB and AI Search.

* adds vector_store configuration to the default config generated by the `--init` flag
* improves current support for managed identities by exposing an optional **audience** parameter in the config file (needed in some use cases for managed identity).
* removes inefficient logic by uploading vector embeddings to the vector store at *index time* and reading from the vector store at *query time*
* removes logic that caused vector embeddings to be uploaded/overwritten to the vector store each time a query is made
* documentation updates

## Related Issues

N/A

## Proposed Changes

The `audience` parameter used to be referred to as `cognitive_services_endpoint` but was in conflict with standard Microsoft sdk naming conventions. This parameter has been refactored to use the more generic `audience` name throughout the codebase.

## Checklist

- [x] I have tested these changes locally.
- [x] I have reviewed the code changes.
- [x] I have updated the documentation (if necessary).
- [ ] I have added appropriate unit tests (if applicable).

## Additional Notes

N/A

### CODE REVIEW

**Comment by AlonsoGuevara** [comment]
Hey Kenny, is this intended to be kept as a Draft?

**Comment by jgbradley1** [comment]
While reviewing this PR, I found some missing documentation for the `vector_store` config settings. Will go ahead and add the missing documentation to this PR since this touches on the AI Search vector store audience parameter as well.

File: graphrag/api/query.py [hunk 1]
```diff
@@ -257,24 +251,20 @@ async def local_search_streaming(
...
-    _entities = read_indexer_entities(nodes, entities, community_level)
-
-    lancedb_dir = Path(config.storage.base_dir) / "lancedb"
-
-    vector_store_args.update({"db_uri": str(lancedb_dir)})
+    # TODO: must update filepath of lancedb (if used) until the new config engine has been implemented
+    # TODO: remove the type ignore annotations below once the new config engine has been refactored
+    vector_store_type = config.embeddings.vector_store["type"]  # type: ignore
+    vector_store_args = config.embeddings.vector_store
+    if vector_store_type == "lancedb":
```
Comment (AlonsoGuevara) [inline graphrag/api/query.py:258]: Isn't there an enum to check for this?

**Review by AlonsoGuevara — APPROVED** [review]

File: graphrag/api/query.py (hunk 1 above)
Comment (jgbradley1) [reply graphrag/api/query.py]: Yes. It's been fixed. Thanks!

**Pull request merged**